src/ol_infrastructure/applications/forum/__main__.py

```diff
@@ -0,0 +1,278 @@
+    .read_text()
+    .replace("DEPLOYMENT", f"{stack_info.env_prefix}"),
+)
+# Register edX Platform AMI for Vault AWS auth
+aws_vault_backend = f"aws-{stack_info.env_prefix}"
+iam.RolePolicyAttachment(
+    f"forum-server-describe-instance-role-policy-{env_name}",
+    policy_arn=policy_stack.require_output("iam_policies")["describe_instances"],
+    role=forum_server_instance_role.name,
+)
+iam.RolePolicyAttachment(
+    f"forum-server-route53-role-policy-{env_name}",
+    policy_arn=policy_stack.require_output("iam_policies")["route53_odl_zone_records"],
+    role=forum_server_instance_role.name,
+)
+forum_server_instance_profile = iam.InstanceProfile(
+    f"forum-server-instance-profile-{env_name}",
+    role=forum_server_instance_role.name,
+    path="/ol-infrastructure/forum-server/profile/",
+)
+edxapp_web_vault_auth_role = vault.aws.AuthBackendRole(
+    "forum-web-ami-ec2-vault-auth",
+    backend=aws_vault_backend,
+    auth_type="iam",
+    role="forum-server",
+    inferred_entity_type="ec2_instance",
+    inferred_aws_region="us-east-1",
+    bound_iam_instance_profile_arns=[forum_server_instance_profile.arn],
+    bound_ami_ids=[forum_server_ami.id],
+    bound_account_ids=[aws_account.account_id],
+    bound_vpc_ids=[vpc_id],
+    token_policies=[forum_vault_policy.name],
+)
+consul_datacenter = consul_stack.require_output("datacenter")
+grafana_credentials = read_yaml_secrets(
+    Path(f"vector/grafana.{stack_info.env_suffix}.yaml")
+)
+block_device_mappings = [BlockDeviceMapping()]
+tag_specs = [
+    TagSpecification(
+        resource_type="instance",
+        tags=aws_config.merged_tags({"Name": forum_server_tag}),
+    ),
+    TagSpecification(
+        resource_type="volume",
+        tags=aws_config.merged_tags({"Name": forum_server_tag}),
+    ),
+]
+forum_server_security_group = ec2.SecurityGroup(
+    f"forum-server-security-group-{env_name}",
+    name=f"forum-server-{target_vpc_name}-{env_name}",
+    description="Access control for forum servers",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            protocol="tcp",
+            from_port=DEFAULT_HTTPS_PORT,
+            to_port=DEFAULT_HTTPS_PORT,
+            cidr_blocks=["0.0.0.0/0"],
+            description=f"Allow traffic to the forum server on port {DEFAULT_HTTPS_PORT}",
+        ),
+    ],
+    egress=default_egress_args,
+    vpc_id=vpc_id,
+)
+lt_config = OLLaunchTemplateConfig(
+    block_device_mappings=block_device_mappings,
+    image_id=forum_server_ami.id,
+    instance_type=forum_config.get("instance_type") or InstanceTypes.burstable_medium,
+    instance_profile_arn=forum_server_instance_profile.arn,
+    security_groups=[
+        forum_server_security_group,
+        consul_security_groups["consul_agent"],
+        target_vpc["security_groups"]["integrator"],
+    ],
+    tags=aws_config.merged_tags({"Name": forum_server_tag}),
+    tag_specifications=tag_specs,
+    user_data=consul_datacenter.apply(
```

Comment:
There is no forum specific userdata that needs to be created? Or we don't know yet?